src/zcash/Note.hpp

```diff
@@ -8,7 +8,12 @@
 
 namespace libzcash {
 
-class Note {
+class BaseNote {
+public:
+    virtual uint256 cm() const {};
```

Comment:
LGTM; commitments for Sapling notes can be deterministically computed from their contents alone.